Add multi-object tracking dabble node (#477)
aimakerspace/PeekingDuck: docs/source/use_cases/multi_object_tracking.md

# Multi Object Tracking

## Overview

> Multi Object Tracking (MOT) is the task of taking an initial set of object detections, creating a unique ID for each of the initial detections, and then tracking each of the objects as they move around frames in a video, maintaining the ID assignment.

Reviewer: ```suggestion
Multi Object Tracking (MOT) is the task of detecting unique objects and tracking them as they move across frames in a video.
```

Author: Will implement this suggestion

There are three main components to obtain the track of an object: 1) object detection using AI; 2) measure similarity between objects in frames; and 3) recover the identity information based on the similarity measurement between objects across frames. In this section, each tracking algorithm will be explained based on its components.

#### 1. Object Detection

The MOT node requires a detected bounding box from an object detector model. To achieve this with PeekingDuck, you may use our open source models such as YOLOv4, EfficientDet, and [PoseNet](https://arxiv.org/abs/1505.07427) (for human detection only) which return detected bounding boxes. This allows the application to identify where each object is located within the video feed. The location is returned as two (x, y) coordinates in the form [x1, y1, x2, y2], where (x1, y1) is the top-left corner of the bounding box, and (x2, y2) is the bottom-right. These are used to form the bounding box of each object detected which will then be used to determine a track for each object.

<img src="https://raw.githubusercontent.com/aimakerspace/PeekingDuck/dev/images/readme/yolo_demo.gif" width="70%">

#### 2. MOSSE (OpenCV)

> Minimum Output Sum of Squared Error (MOSSE) uses an adaptive correlation for object tracking which produces stable correlation filters when initialized using a single frame. MOSSE tracker is robust to variations in lighting, scale, pose, and non-rigid deformations. It also detects occlusion based upon the peak-to-sidelobe ratio, which enables the tracker to pause and resume where it left off when the object reappears. MOSSE tracker also operates at a higher fps. To add to the positives, it is also very easy to implement, is as accurate as other complex trackers and much faster. But, on a performance scale, it lags behind the deep learning based trackers.

Author: Will implement this suggestion

> The bounding boxes detected in the first frame are used to initialize a single tracker instance for each detection. The tracker for each bounding box is then updated per frame and is deleted when the tracker fails. 

Author: Will implement this suggestion
